src/main/java/com/codeborne/selenide/commands/ancestor/AncestorWithAttributeAndValueRule.java

```diff
@@ -0,0 +1,25 @@
+package com.codeborne.selenide.commands.ancestor;
+
+import static java.lang.String.format;
+
+public class AncestorWithAttributeAndValueRule extends SelectorValidation implements AncestorRule {
+
+  private String xpath;
+
+  @Override
+  public boolean evaluate(String selector, int index) {
+    boolean evaluationResult = false;
+    if (isAttribute(selector) && containsAttributeValue(selector)) {
+      String attribute = selector.substring(1, selector.indexOf("="));
```

Comment:
For better performance, calculate `selector.indexOf("=")` only once a put to a local variable.

Reply:
Thanks, updated.